app/helpers/markdown_helper.rb

```diff
@@ -4,13 +4,41 @@ module MarkdownHelper
   def camofy(markdown)
     return unless markdown
 
-    markdown.gsub(markdown_img_regex) do
+    markdown = sub_markdown(markdown)
+    sub_html(markdown)
+  end
+
+  def sub_markdown(text)
+    text.gsub(markdown_img_regex) do
       "![#{Regexp.last_match(1)}](#{camo(Regexp.last_match(2))}#{Regexp.last_match(3)})"
     end
   end
 
+  def sub_html(text)
+    text.gsub(html_img_regex) do
+      preceding_src = Regexp.last_match(1)
+      quote_mark = Regexp.last_match(2)
+      url = Regexp.last_match(3)
+      ending = Regexp.last_match(4)
+      "<img#{preceding_src} src=#{quote_mark}#{camo(url)}#{quote_mark}#{ending}"
+    end
+  end
+
   def markdown_img_regex
     # ![alt text](url =widthxheight "title")
     /!\[([^\[]*)\]\(([^\ )]+)(( =[^)]?[^ ]+)?( [^)]?"[^)]+")?)?\)/
   end
+
+  def html_img_regex
+    # warning: this regex may not be perfect. They rarely are.
+    # If you find an edge case, improve this regex!
+    # <img...something... src="url"...ending...
+    # or, the alternative quotes: <img...something... src='url'...ending...
+    # or, even without quotes: <img...something... src=url...ending...
+    # and the ...ending... can be either a space, a > or />
+    # note that we don't allow mismatched quotes like 'url" or shenanigans like that
+    # This regex contains two particularly useful features:
+    #  capturing groups, and lazy matching.
+    %r{<img([^>]*) src=(["']?)(.+?)\2( |>|/>)}
```

Comment:
don't ask me why I had to put %r{} around it, I just followed rubocop's instructions.

Reply:
is there any documentaion online that i can use as a guide line for the review?

Reply:
If you've never seen regexes before, they are a pain in the ass to review. They are kind of like a language in their own right, and the main way to understand regexes is to play around building them and testing them out on a site that shows you what the regex matches.
I've taken the regex here and prepared an example for you: https://regex101.com/r/RLd8UL/1
If you'd like to have a quick online meeting about how to understand this, let me know. I have time tomorrow

Reply:
I have seen regex before, and it is almost unreadable. thank you for the resource

Reply:
The best way to test the regex is probably to come up with valid HTML img tags that are not matched by the regex. The edge cases I could think of have been captured in this one.

Reply:
I am cheating a little bit by using chatgpt, it has some suggestions and explaination, i will check it and comment it